plugins/modules/swim_workflow_manager.py

```diff
@@ -1064,119 +1255,158 @@ def get_diff_import(self):
+                            break
+                        else:
+                            self.status = "failed"
+                            self.msg = task_details.get("failureReason", "SWIM Image {0} seems to be invalid".format(image_name))
+                            self.log(self.msg, "WARNING")
+                            self.result['response'] = self.msg
+                            return self
+
+                self.result['response'] = task_details if task_details else response
+
+                image_name = image_name.split('/')[-1]
+                image_id = self.get_image_id(image_name)
+                self.have["imported_image_id"] = image_id
 
-                if task_details and task_details.get("isError"):
-                    if "already exists" in task_details.get("failureReason", ""):
-                        self.msg = "SWIM Image {0} already exists in the Cisco Catalyst Center".format(name)
-                        self.result['msg'] = self.msg
-                        self.log(self.msg, "INFO")
-                        self.status = "success"
-                        self.result['changed'] = False
-                        break
-                    else:
-                        self.status = "failed"
-                        self.msg = task_details.get("failureReason", "SWIM Image {0} seems to be invalid".format(image_name))
-                        self.log(self.msg, "WARNING")
-                        self.result['response'] = self.msg
-                        return self
+            imported_images_str = ", ".join(images_to_import)
+            skipped_images_str = ", ".join(existing_images)
 
-            self.result['response'] = task_details if task_details else response
+            if skipped_images_str and not imported_images_str:
+                self.msg = "Image(s) {0} skipped as they already exist in Cisco Catalyst Center. No images were imported.".format(skipped_images_str)
+            elif skipped_images_str and imported_images_str:
+                self.msg = ("Image(s) {0} skipped as they already exist Cisco Catalyst Center. Images {1} have been imported"
+                            " successfully.").format(skipped_images_str, imported_images_str)
+            else:
+                self.msg = "Image(s) {0} have been imported successfully in Cisco Catalyst Center.".format(imported_images_str)
 
-            # Fetch image_id for the imported image for further use
-            image_name = image_name.split('/')[-1]
-            image_id = self.get_image_id(image_name)
-            self.have["imported_image_id"] = image_id
+            self.result['msg'] = self.msg
+            self.result['response'] = self.msg
+            self.log(self.msg, "INFO")
 
             return self
 
         except Exception as e:
             self.status = "failed"
             self.msg = """Error: Import image details are not provided in the playbook, or the Import Image API was not
```

Comment:
self.msg = (
        "Error: Import image details are missing from the playbook or the Import Image API was not "
        "triggered successfully. Please ensure that all necessary details are provided and verify the "
        "status of the Import Image process. Details: {0}".format(str(e))
    )

Reply:
updated the code